src/gtc/dace/utils.py

```diff
@@ -638,8 +742,10 @@ def node_match(n1, n2):
                 assert isinstance(n2, dace.nodes.AccessNode)
                 assert n1.access == n2.access
                 assert n1.data == n2.data
-            elif isinstance(n1, OIRLibraryNode):
-                assert n1 == n2
+            elif isinstance(n1, VerticalLoopLibraryNode):
+                assert equal_vl_node(n1, n2)
+            elif isinstance(n1, HorizontalExecutionLibraryNode):
+                assert equal_he_node(n1, n2)
```

Comment:
same here, do we really want to use assertions?

Reply:
fixed